tests/lib/openstack_query/handlers/test_client_side_handler_datetime.py

```diff
@@ -0,0 +1,158 @@
+import unittest
+from datetime import datetime
+from unittest.mock import patch
+from parameterized import parameterized
+
+from openstack_query.handlers.client_side_handler_datetime import (
+    ClientSideHandlerDateTime,
+)
+
+from tests.lib.openstack_query.mocks.mocked_props import MockProperties
+
+from enums.query.query_presets import QueryPresetsDateTime
+
+
+@patch(
+    "openstack_query.time_utils.TimeUtils.get_current_time",
+    return_value=datetime(2023, 6, 4, 10, 30, 0),
+)
+class ClientSideHandlerDateTimeTests(unittest.TestCase):
+    """
+    Run various tests to ensure that ClientSideHandlerDateTime class methods function expectedly
+    """
+
+    def setUp(self):
+        """
+        Setup for tests
+        """
+        super().setUp()
+        # sets filter function mappings so that PROP_1 is valid for all client_side
+        _FILTER_FUNCTION_MAPPINGS = {
+            preset: [MockProperties.PROP_1] for preset in QueryPresetsDateTime
+        }
+        self.instance = ClientSideHandlerDateTime(_FILTER_FUNCTION_MAPPINGS)
+
+        self.test_cases = {
```

Comment:
Yea it's a little confusing 

We set the current time as 2023-06-04 10:30 AM
We then calculate a relative timestamp from the current time - based on `days`, `hours`, `minutes`, `seconds`
So if we give `days=1` - we expect the relative timestamp to be 2023-06-03 10:30AM

Then we just compare that against `prop` and see if the `prop` timestamp is older or younger than the `relative` timestamp

This just allows you to put in `days=60` for example and find all servers older than 60 days (since prop would be the timestamp each server was created)

Reply:
The names for the test e.g. `older_than_1_day` is what we expect the difference to be between the `prop` timestamp and the calculated `relative` timestamp

Reply:
So this effectively gives us fixed points to backwards from then is my understanding